user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala

```diff
@@ -222,14 +251,15 @@ class UserManagementService(userRepository: UserRepository, getUserResourceFile:
                       case _ =>
                         Unexpected(s"Wrong formatting : ${file.path}").fail
                     }
+      _          <- userService.reloadPure()
     } yield user
   }
 
   /*
    * When we delete an user, it can be from file or auto-added by OIDC or other backend supporting that.
-   * So we need to both remove it (mark "status=delete") from base and from file.
+   * So we let the callback on file reload do the deletion for file users, and then delete from the user repository.
```

Comment:
This is a tricky bug that relates to how the "reload" work, the callback behind the deletion does some logic on users managed by file that can also change other users.
We absolutely don't want this logic for users managed by other providers, and we wait for the callbacks to terminate before effectively setting the status to deleted.